packages/iov-socket/src/queueingstreamingsocket.spec.ts

```diff
@@ -87,4 +89,51 @@ describe("QueueingStreamingSocket", () => {
       socket.reconnect();
     });
   });
+
+  describe("connectionStatus", () => {
+    it("exposes connection status", done => {
+      pendingWithoutSocketServer();
+      const socket = new QueueingStreamingSocket(socketServerUrl);
+      let statusChangesSeen = 0;
+      socket.connectionStatus.updates.subscribe({
+        next: status => {
+          if (status === null) {
+            return;
+          }
+          switch (statusChangesSeen++) {
+            case 0:
+            case 4:
+              expect(status).toEqual(WebSocket.CONNECTING);
+              break;
+            case 1:
+            case 5:
+              expect(status).toEqual(WebSocket.OPEN);
+              break;
+            case 2:
+            case 6:
+              expect(status).toEqual(WebSocket.CLOSING);
+              break;
+            case 3:
+            case 7:
+              expect(status).toEqual(WebSocket.CLOSED);
+              break;
+            default:
+              fail("Got too many status changes");
```

Comment:
when there is a `done` callback, you need to use `done.fail`. The simple `fail` is unreliable in this case.